### PR TITLE
Old style interface should warn to stderr by default

### PR DESCRIPTION
**Pull request by jamesls**
I noticed when I was trying out the old interface that the `DeprecationWarning` does *not* print to stderr.  We really should be at the point out where the error is front and center whenever you try to do anything with the old interface.  This change does that.

In order to help with migration I also sketched out a migration guide for going from the old interface to the new client interface so we have something we can point customer to when they run into issues.

I also manually verified the errors are on stderr now:

```
>>> import botocore.session
>>> s = botocore.session.get_session()
>>> s.get_service('ec2')
botocore/session.py:560: ImminentRemovalWarning: get_service is deprecated and will be removed.  Use create_client instead.
  "Use create_client instead.", ImminentRemovalWarning)
botocore/service.py:231: ImminentRemovalWarning: get_service is deprecated and will be removed.  Use Session.create_client instead.
  ImminentRemovalWarning)
Service(ec2)
```

cc @kyleknap @danielgtaylor

### CODE REVIEW

File: docs/source/client_upgrades.rst [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    ec2 = session.get_client('ec2', 'us-west-2')
+    parsed = ec2.describe_instances()
+    print("The status code is:",
+          parsed['ResponseMetadata']['HTTPStatusCode']))
+
+
+Error Handling
+--------------
+
+With clients, exceptions are now raised on any non 200 response.
```
Comment (kyleknap) [inline docs/source/client_upgrades.rst]: I feel that more specifically it is any non 2xx response.

File: docs/source/client_upgrades.rst [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+**New**
+
+.. code-block:: python
+
+    from botocore.exceptions import ClientError
+
+    ec2 = session.get_client('ec2', 'us-west-2')
+    try:
+        parsed = ec2.describe_instances()
```
Comment (kyleknap) [inline docs/source/client_upgrades.rst]: Will this call throw a ResourceUnavailable error? From my experience, this call works for me without errors nearly all of the time.

**Comment by kyleknap** [comment]
Looks good. I had a couple comments. This will help a lot if a user upgrades when we do the alpha release and has not switched to clients. :ship:

File: docs/source/client_upgrades.rst (hunk 2 above)
Comment (jamesls) [inline docs/source/client_upgrades.rst]: I would assume it's possible, even if unlikely.  If there's another call you'd prefer I can switch it.

File: docs/source/client_upgrades.rst (hunk 1 above)
Comment (jamesls) [inline docs/source/client_upgrades.rst]: Fixed.

File: docs/source/client_upgrades.rst (hunk 2 above)
Comment (kyleknap) [inline docs/source/client_upgrades.rst]: @jamesls 
Can we just make it a call to describe_instances using an instance-id of a non-existent instance?

Comment (jamesls) [inline docs/source/client_upgrades.rst]: Sure, I'll update.

**Comment by jamesls** [comment]
Cool, I believe all the feedback has been incorporated.

**Comment by kyleknap** [comment]
Thanks! Good to merge.

**Pull request merged**